import "session_commands.proto";
import "game_commands.proto";
import "room_commands.proto";
import "moderator_commands.proto";
import "admin_commands.proto";

message CommandContainer {
	enum CommandType {
		SESSION = 0;
		GAME = 1;
		ROOM = 2;
		MODERATOR = 3;
		ADMIN = 4;
	}
	required uint64 cmd_id = 1;
	
	optional uint32 game_id = 10;
	optional uint32 room_id = 20;
	
	repeated SessionCommand session_command = 100;
	repeated GameCommand game_command = 101;
	repeated RoomCommand room_command = 102;
	repeated ModeratorCommand moderator_command = 103;
	repeated AdminCommand admin_command = 104;
}
